src/components/calcite-picker/calcite-picker-row/calcite-picker-row.tsx

```diff
@@ -0,0 +1,143 @@
+import { Component, Element, Event, EventEmitter, Host, Method, Prop, h } from "@stencil/core";
+import {
+  checkSquare16,
+  circle16,
+  circleFilled16,
+  handleVertical24,
+  square16,
+  trash16
+} from "@esri/calcite-ui-icons";
+import { CSS } from "./resources";
+import CalciteIcon from "../../_support/CalciteIcon";
+
+@Component({
+  tag: "calcite-picker-row",
+  styleUrl: "./calcite-picker-row.scss",
+  shadow: true
+})
+export class CalcitePickerRow {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop({ reflect: true }) textHeading: string;
+
+  @Prop({ reflect: true }) textDescription: string;
+
+  @Prop({ reflect: true }) value: string;
+
+  @Prop({ reflect: true, mutable: true }) selected = false;
+
+  @Prop({ reflect: true }) icon: "square" | "circle" | "grip" | null = null;
+
+  @Prop({ reflect: true }) editing = false;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() rowToggled: EventEmitter;
+  @Event() rowDeleted: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggle() {
+    this.selected = !this.selected;
+    this.rowToggled.emit({ row: this.el, value: this.value, selected: this.selected });
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  deleteRow() {
+    this.rowDeleted.emit({ row: this.el, value: this.value });
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  renderIcon() {
+    if (!this.icon) {
+      return null;
+    }
+    if (this.icon === "grip") {
+      return (
+        <span class="handle">
+          <CalciteIcon size="24" path={handleVertical24} />
+        </span>
+      );
+    } else {
+      /* tslint:disable */
```

Comment:
So as far as moving forward - do you guys agree with my judgement that this is more readable than the alternative? If so can we explore the tslint rule soonish? Otherwise I'll have to change it, then change it back.